srtcore/utilities.h

```diff
@@ -548,6 +548,28 @@ class UniquePtr: public std::auto_ptr<T>
 
 #endif
 
+template <class Signature>
+struct CallbackHolder
+{
+    void* opaque;
+    Signature* fn;
+
+    CallbackHolder(): opaque(NULL), fn(NULL)  {}
+
+    void set(void* o, Signature* f)
+    {
+        // Test if the pointer is a pointer to function. Don't let
+        // other type of pointers here.
+        void* (*testfn)(void*) ATR_UNUSED = (void*(*)(void*))f;
```

Comment:
Does this expression have a meaning if you are explicitly casting the pointer?

Reply:
Yes. Just I'm casting between two different function pointers, which is allowed in the standard, unlike casting between function pointer and data pointer. Compilers treat this various ways, but most should complain here about incorrect pointer cast. This is in order to try to pop up the diagnostic in case when someone used a data pointer here.

Reply:
I think this is overkill, to be honest. We are not creating a universal SDK here, - this is internal functionality. But since no harm is done, I won't complain (any further) if you decide to leave this check.